include/common/serialise_matrix.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+// BoB robotics includes
+#include "macros.h"
+
+// Third-party includes
+#include "third_party/path.h"
+
+// Eigen
+#include <Eigen/Core>
+
+// Standard C++ includes
+#include <fstream>
+
+namespace BoBRobotics {
+//! Read a matrix of type T from a binary file
+template<class T>
+auto
+readMatrix(const filesystem::path &filepath)
+{
+    // Open file
+    std::ifstream ifs;
+    ifs.exceptions(std::ios::badbit | std::ios::eofbit);
+    ifs.open(filepath.str(), std::ios::in | std::ios::binary);
```

Comment:
**really** not important but ``std::ifstream`` always |s ``std::ios::in`` with the flags that are passed (weird behaviour but still) so it's not necessary to repeat it

Reply:
Heh TIL. C++ is odd.